src/coreclr/jit/gentree.cpp

```diff
@@ -19323,6 +19325,12 @@ GenTree* Compiler::gtNewSimdCmpOpAllNode(genTreeOps  op,
 
             if (simdSize == 32)
             {
+                // TODO-XArch-CQ: It's a non-trivial amount of work to support these
+                // for floating-point while only utilizing AVX. It would require, among
+                // other things, inverting the comparison and potentially support for a
+                // new Avx.TestNotZ intrinsic to ensure the codegen remains efficient.
+                assert(compIsaSupportedDebugOnly(InstructionSet_AVX2));
+
```

Comment:
Just noting that this is worth doing even if we decided to only accelerate on AVX2+. Using `CompareNotGreaterThan` allows us to use `PTEST`, which in turn is more efficient and allows the code to directly branch rather than requiring a `MoveMask` instruction.